tools/distpartitioning/utils.py

```diff
@@ -8,29 +8,38 @@
 import pyarrow
 from pyarrow import csv
 
-def read_partitions_file(part_file):
+def read_ntype_partition_files(schema_map, input_dir):
     """
-    Utility method to read metis partitions, which is the output of 
-    pm_dglpart2
+    Utility method to read the partition id mapping for each node.
+    For each node type, there will be an file, in the input directory argument
+    containing the partition id mapping for a given nodeid. 
 
     Parameters:
     -----------
-    part_file : string
-        file name which is the output of metis partitioning
-        algorithm (pm_dglpart2, in the METIS installation).
-        This function expects each line in `part_file` to be formatted as 
-        <global_nid> <part_id>
-        and the contents of this file are sorted by <global_nid>. 
+    schema_map : dictionary
+        dictionary created by reading the input metadata json file
+    input_dir : string
+        directory in which the node-id to partition-id mappings files are 
+        located for each of the node types in the input graph
 
     Returns:
     --------
-    numpy array
-        array of part_ids and the idx is the <global_nid>
+    numpy array : 
+        array of integers representing mapped partition-ids for a given node-id
+        the line number is used as the type_node_id in each of the files.
     """
-    partitions_map = np.loadtxt(part_file, delimiter=' ', dtype=np.int64)
-    #as a precaution sort the lines based on the <global_nid>
-    partitions_map = partitions_map[partitions_map[:,0].argsort()]
-    return partitions_map[:,1]
+    assert os.path.isdir(input_dir)
+
+    #iterate over the node types and extract the partition id mappings
+    part_ids = []
+    ntype_names = schema_map[constants.STR_NODE_TYPE]
+    for ntype in ntype_names:
+        df = csv.read_csv(os.path.join(input_dir, '{}.txt'.format(ntype)), \
+                read_options=pyarrow.csv.ReadOptions(autogenerate_column_names=True), \
+                parse_options=pyarrow.csv.ParseOptions(delimiter=' '))
+        ntype_partids = df['f0'].to_numpy()
+        part_ids.append(ntype_partids)
+    return np.concatenate(part_ids)
```

Comment:
it seems to me that the returned partition IDs are stored for all nodes in the graph.

Reply:
Yes. This is the for the entire graph. Currently this is a bottleneck. Per our discussion, we have to change this so that each node only stores the partition-ids for the nodes which it owns. This is targetted for the next version of the release.